File: TravellerCreatorModels/StarFrontiers/StarFrontiersHex.cs
using TravellerCreatorModels.Interfaces;
using TravellerCreatorModels.Other;

namespace TravellerCreatorModels.StarFrontiers;

public class StarFrontiersHex : IHex
{
    public Coordinates Coordinates { get; set; }
    public List<IStarSystem> StarSystems { get; set; } = new();

    public StarFrontiersHex(Coordinates coordinates)
    {
        Coordinates = coordinates;
    }
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/star-frontiers/star-frontiers-hex.ts
import {IHex} from "../interfaces/hex";
import {Coordinates} from "../other/coordinates";
import {StarFrontiersStarSystem} from "./star-frontiers-star-system";

export class StarFrontiersHex implements IHex {
  coordinates: Coordinates;
  starSystems: StarFrontiersStarSystem[];
}
